packages/server-wallet/src/channel-manager.ts

```diff
@@ -0,0 +1,92 @@
+import {CreateChannelParams} from '@statechannels/client-api-schema';
+import {backOff, IBackOffOptions} from 'exponential-backoff';
+
+import {MessageServiceInterface} from './message-service/types';
+import {getMessages} from './message-service/utils';
+import {WalletObjective} from './models/objective';
+import {Wallet} from './wallet';
+export const delay = async (ms = 10): Promise<void> =>
+  new Promise(resolve => setTimeout(resolve, ms));
+
+export class ChannelManager {
+  /**
+   * Constructs a channel manager that will ensure objectives get accomplished by resending messages if needed.
+   * @param wallet The wallet to use.
+   * @param messageService  The message service to use.
+   * @param backOffIntervals At what intervals ensureObjective should retry sending messages to complete an objective.
+   * @returns A channel manager.
+   */
+  public static async create(
+    wallet: Wallet,
+    messageService: MessageServiceInterface,
+    backOffOptions?: Partial<Exclude<IBackOffOptions, 'retry'>>
+  ): Promise<ChannelManager> {
+    return new ChannelManager(wallet, messageService, backOffOptions ?? {});
+  }
+  private constructor(
+    private _wallet: Wallet,
+    private _messageService: MessageServiceInterface,
+    private _backOffOptions: Partial<IBackOffOptions>
+  ) {}
+
+  /**
+   * TODO: This is a basic implementation of createChannels.
+   * This will be cleaned up in https://github.com/statechannels/statechannels/issues/3365
+   * @param args
+   * @param numberOfChannels
+   * @returns
+   */
+  public async createChannels(args: CreateChannelParams, numberOfChannels: number): Promise<void> {
+    const createResult = await this._wallet.createChannels(args, numberOfChannels);
+
+    await this._messageService.send(getMessages(createResult));
+    await this.ensureObjectives(createResult.newObjectives);
```

Comment:
What if `this._messageService.send(getMessages(createResult))` results in the objectives succeeding?

Reply:
That's a good point! We should probably filter out any successful objectives when `ensureObjectives` is called.

Reply:
The alternative to that is to pass the entire `createResult` to `ensureObjectives`, and send the initial messages from within that. I think that alternative is simpler?

Reply:
Done in e389707